onadata/apps/api/viewsets/open_data_viewset.py

```diff
@@ -225,6 +125,153 @@ def append_to_tableau_colulmn_headers(header, question_type=None):
 
         return tableau_colulmn_headers
 
+    def fetch_form_submissions(self, xform, gt_id=None, count_check=False):
+        if xform.is_merged_dataset:
+            qs_kwargs = {'xform_id__in': list(
+                xform.mergedxform.xforms.values_list('pk', flat=True))}
+        else:
+            qs_kwargs = {'xform_id': xform.pk}
+        if gt_id:
+            qs_kwargs.update({'id__gt': gt_id})
+        instances = Instance.objects.filter(
+            **qs_kwargs, deleted_at__isnull=True).order_by('pk')
+
+        if count_check:
+            xform_json = json.loads(xform.json)
+            fields = xform_json.get('children')
+            # Capturing fields of type repeat at zero level on the form
+            repeat_fields = [
+                d['name'] for d in fields if d['type'] == 'repeat']
+
+            # Figure a way to get field path for nested repeat
+            # as the name field cannot resonate on the submission data fields
+
+            # for d in fields:
+            #     try:
+            #         if d['children']:
+            #             for x in d['children']:
+            #                 if x['type']  == 'repeat':
+            #                     repeat_fields.append(x['name'])
+            #     except (KeyError, NameError):
+            #         continue
+            repeat_subs_len = []
+            for field in repeat_fields:
+                for instance in instances:
+                    val = len(instance.json[field])
+                    repeat_subs_len.append({
+                        f'field_count': val,
+                        f'instance_id': instance.pk,
+                        f'repeat_field': field})
+
+            # This is an asumption also that there
+            # is only 1 repeat group defined on the xform.
+            # Another assumption is that the repeat count
+            # at zero level of the form is the
+            # same as the nested repeat count
+            largest_instance = max(
+                repeat_subs_len, key=lambda x: x['field_count'])
+            largest_repeat_count = largest_instance['field_count']
+
+            return largest_repeat_count
+
+        return instances
+
+    def process_tableau_data(self, data, xform):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/gh/onaio/onadata/pullRequest?prid=6116910)